compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -5581,18 +5631,18 @@ private BType resolveQueryType(SymbolEnv env, BLangExpression selectExp, BType c
         }
 
         if (selectTypes.size() == 1) {
-            BType errorType = getErrorType(collectionType, queryExpr, data);
+            BType completionType = getCompletionType(collectionType, queryExpr.isStream, data);
```

Comment:
This may not related to this PR, can we maintain like one enum instead of` isStream, isTable`.. etc